custom/ilsgateway/tanzania/handlers/zipline.py

```diff
@@ -0,0 +1,64 @@
+from corehq.apps.products.models import SQLProduct
+from custom.ilsgateway.tanzania.handlers.keyword import KeywordHandler
+from custom.ilsgateway.tanzania.reminders import INVALID_PRODUCT_CODE
+from custom.zipline.api import ProductQuantity
+
+
+class ParseError(Exception):
+    pass
+
+
+class ZiplineGenericHandler(KeywordHandler):
+
+    help_message = None
+    error_message = None
+
+    def _check_quantities(self, quantities):
+        for quantity in quantities:
+            try:
+                int(quantity)
+            except ValueError:
+                raise ParseError(self.error_message)
+
+    def _check_product_codes(self, product_codes):
```

Comment:
(i think this method is ok for now but we'll likely need to update this once the requirements become clearer on how to define the restricted set of products)